src/util/tempdir.cpp

```diff
@@ -34,26 +34,30 @@ std::string get_temporary_directory(const std::string &name_template)
   std::string result;
 
   #ifdef _WIN32
-    DWORD dwBufSize = MAX_PATH;
-    char lpPathBuffer[MAX_PATH];
+    DWORD dwBufSize = MAX_PATH+1;
+    char lpPathBuffer[MAX_PATH+1];
```

Comment:
The documentation of `GetTempFileName` says that it is a precondition that the path not be longer than `MAX_PATH–14`. Hence those changes seem unnecessary.

Reply:
Yes; but that may be subtle. The documentation of GetTempPathA says that the maximal return value is MAX_PATH+1; this is the length of the buffer required.